src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/MigrationService.kt

```diff
@@ -113,6 +113,11 @@ abstract class MigrationService<FILTER : Any, NOMIS_ID : Any, NOMIS_ENTITY : Any
       )
     } else {
       if (context.body.hasCheckedAReasonableNumberOfTimes(completeCheckCount)) {
+        migrationHistoryService.recordMigrationCompleted(
+          migrationId = context.migrationId,
+          recordsFailed = queueService.countMessagesThatHaveFailed(context.type),
+          recordsMigrated = getMigrationCount(context.migrationId),
+        )
```

Comment:
I don't think it matters if we record history or publish telemetry first? I want to publish telemetry last because it's a nice marker that an intregration test has finished.